google-cloud-datastore/src/main/java/com/google/cloud/datastore/RecordQuery.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.datastore;
+
+import com.google.api.core.InternalApi;
+import com.google.cloud.datastore.Query.ResultType;
+
+@InternalApi
+public interface RecordQuery<V>{
```

Comment:
is this still needed?

Reply:
Hi @kolea2 , before this feature, `Query` (top level class) was used heavily in the code to generalise between GqlQuery and StructuredQuery,  We used `Query` reference so that different components can still remain oblivious to the underlying Query. 

With the arrival of AggregationQuery, we had to move entity records specific behaviour to `RecordQuery` to keep the `Query` clean enough to be the top level class. Instead of using `Query` reference, existing query handling mechanism uses `RecordQuery` so that existing code doesn't change much.

If we dont do this, then we need to write code to handle StructuredQuery and GqlQuery separately.